Makefile

```diff
@@ -64,20 +64,24 @@ pull: ## update the Docker image used by "make shell"
 	docker pull edxops/edxapp:latest
 
 pre-requirements: ## install Python requirements for running pip-tools
+	pip install -qr requirements/pip.txt
 	pip install -qr requirements/edx/pip-tools.txt
 
 local-requirements:
 # 	edx-platform installs some Python projects from within the edx-platform repo itself.
 	pip install -e .
 
-dev-requirements: local-requirements
-	pip install -qr requirements/edx/development.txt
+dev-requirements: pre-requirements
+	pip-sync requirements/edx/development.txt
+	make local-requirements
 
-base-requirements: local-requirements
-	pip install -qr requirements/edx/base.txt
+base-requirements: pre-requirements
+	pip-sync requirements/edx/base.txt
+	make local-requirements
 
-test-requirements: local-requirements
-	pip install --exists-action='w' -qr requirements/edx/testing.txt
+test-requirements: pre-requirements
+	pip-sync --pip-args="--exists-action=w" requirements/edx/testing.txt
+	make local-requirements
 
 requirements: pre-requirements ## install development environment requirements
```

Comment:
Looks like we have some similar but divergent logic between `requirements` and `dev-requirements`. Could you:
* add the `wildcard requirements/edx/private.txt` stuff that is here to `dev-requirements`, and then
* make this a simple alias to `dev-requirements`?